SCAutolib/models/sssd_conf.py

```diff
@@ -0,0 +1,159 @@
+"""
+This module provide information and methods to create and modify
+/etc/sssd/sssd.conf file. Following methods are implemented:
+
+create: create content of internal parser object representing sssd.conf
+set:    set or modify content of internal parser object representing sssd.conf
+save:   save content of internal parser object representing sssd.conf to
+        /etc/sssd/sssd.conf file
+clean:  restore original state of /etc/sssd/sssd.conf file
+"""
+
+from configparser import ConfigParser
+from pathlib import Path
+from shutil import copy2
+from typing import Union
+
+from SCAutolib import TEMPLATES_DIR, LIB_BACKUP
+from SCAutolib import logger
+from SCAutolib.models.file import File
+
+
+class SSSDConf(File):
+    """
+    This class contains information and methods to handle sssd.conf file
+
+    It is implemented as singleton, which allows to use class object
+    _default_parser as representation of content of sssd.conf file during
+    runtime.
+
+    Intended use is to create/update and save config file in first runtime
+    and load content of config file to internal parser object in in following
+    runtimes.
```

Comment:
How is sssd.conf different from softhsm2.conf to have the class like this?

Reply:
> How is sssd.conf different from softhsm2.conf to have the class like this?

There are two reasons why sssd.conf is implemented this way:
1) It's system file in a sense that location of this file is known to the system without our library. Paths to other config files are provided by library and are somehow 'internal' to our use. This makes handling of sssd.conf a bit specific (i.e. backups etc.)
2) We modify sssd.conf often during the testing and therefore it is beneficial to store its content in an object that is present during whole runtime and represents sssd.conf. That's main reason for having singleton class. Configs that are not modified often or are not modified at all after they are created does not need this property.

Reply:
Right, I don't mind that it is a singleton class, but isn't that true for Softhsm.conf for example? In a sense that we know its location and we need to be sure if it exists we need to back it up and then restore? I understand that softhsm won't be modified a lot, but we should probably be consistent in how we handle system config files unless we have a very good reason not to.

Reply:
> we should probably be consistent in how we handle system config files unless we have a very good reason not to.

I will think about proper solution and get back to you to consult.